src/cmd/cli/command/compose.go

```diff
@@ -79,8 +79,8 @@ func makeComposeUpCmd() *cobra.Command {
 
 			numGPUS := compose.GetNumOfGPUs(cmd.Context(), project)
 			if numGPUS > 0 {
-				if err := gating.HasAuthorization(gating.ResourceGPU, "usage of GPUs. To resolve see https://docs.defang.io/docs/tutorials/deploy-with-gpu"); err != nil {
-					return err
+				if !gating.Gates.Gpu {
+					return gating.ErrNoPermission("usage of GPUs. To resolve see https://docs.defang.io/docs/tutorials/deploy-with-gpu")
```

Comment:
The tutorial doesn't mention anything about tiers or gating or permissions, so I could go follow the tutorial and still see the same error.

I feel like the error message needs to say something actionable, like:
```
Your tier does not have permission to deploy with GPUs. Upgrade to the next tier at https://portal.defang.dev/pricing
```

This still makes me cringe a bit, because it feels like we've introduced this abstraction of gates to avoid having to check account tiers directly in the CLI, but we can't tell the user "the gate is closed" (since they don't know what that means). So we have to sidestep our abstraction and mention account tiers in the error message because its the only thing that is actionable.

Better than sending them to an article that doesn't actually answer their question though. : /